### ISSUE TITLE
Rename generated dist/auth0-vue.production.esm.js to dist/auth0-vue.production.esm.mjs

### ISSUE THREAD

**Issue by joostvdwsd** [auth0/auth0-vue#429]
### Checklist

- [X] The issue can be reproduced in the [auth0-vue sample app](https://github.com/auth0-samples/auth0-vue-samples/tree/master/01-Login) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-vue#readme), [Examples](https://github.com/auth0/auth0-vue/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-vue/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/auth0-vue/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-vue/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

I'm trying to use Astro + Vue in a yarn berry environment. 

The auth0-vue package is working perfect in runtime vue but not during the build phase.

While building I get the error "Cannot use import statement outside a module". This is fixed by either:
- adding type: module to the package.json of @auth0/auth0-vue
- renaming the esm distibutable js so it has the mjs extension (dist/auth0-vue.production.esm.js => dist/auth0-vue.production.esm.mjs)

The build breaks on the import statement itself so wrapping this in the normal "onMounted" will not work.
 
Basically the generated package is incorrect. Therefor I see this as a bug.

### Reproduction

At this moment I do not have a straight forward reproducible example. If this is still needed for this ticket and if I decide to continue with auth0 I might consider making one.

### auth0-vue version

2.3.3

### Vue version

3.5.12

### Which browsers have you tested in?

Chrome, Safari, Firefox
